src/app/course/challenge/course-challenge-create-edit/course-challenge-create-edit.component.ts

```diff
@@ -0,0 +1,130 @@
+    difficulties: Difficulty[]
+    limits: number[]
+
+    list = [{id: 1, name: 'name A'}, {id: 2, name: 'name B'}, {id: 3, name: 'name C'}]
+
+    constructor(
+        private route: ActivatedRoute,
+        private router: Router,
+        private courseEventService: CourseEventService,
+        private readonly categoryService: CategoryService,
+        private readonly difficultyService: DifficultyService,
+        private readonly notificationsService: TuiNotificationsService,
+    ) { }
+
+    ngOnInit(): void {
+        this.courseId = +this.route.snapshot.parent.paramMap.get('courseId')
+        this.challengeForm = CourseEventForm.createChallengeForm()
+
+        if (this.route.snapshot.paramMap.get('eventId')){ // For editing existing challenge, grab the eventId
+            this.eventId = +this.route.snapshot.paramMap.get('eventId')
+            this.courseEventService.getCourseEvent(this.eventId).subscribe( event => {
+                this.event = event
+                this.challengeForm = CourseEventForm.createChallengeFormWithData(this.event)
+            })
+        }
+
+        this.courseEventService.getChallengeTypes().subscribe(
+            response => this.localChallengeTypes = response
+        )
+        this.categoryService.getCategories().subscribe(
+            categories => this.categories = categories
+        )
+        this.difficultyService.getDifficulties().subscribe(
+            difficulties => this.difficulties = difficulties
+        )
+    }
+
+    getChallengeQuestionSets(): FormArray {
+        return this.challengeForm.get('challengeQuestionSets') as FormArray
+    }
+
+    getChallengeQuestionSetFormControls(): FormControl[] {
+        return this.getChallengeQuestionSets().controls as FormControl[]
+    }
+
+    getFormControl(fc: FormControl, field:string): FormControl {
+        return fc.get(field) as FormControl
+    }
+
+    //TODO: Need support from backend
+    getNumQuestionsLimit(fc: FormControl): number {
+        return 5
+    }
+
+    addChallengeQuestionSet() {
+        this.getChallengeQuestionSets().push(CourseEventForm.createChallengeQuestionSetForm())
+    }
+
+    removeChallengeQuestionSet(index: number): void {
+        this.getChallengeQuestionSets().removeAt(index)
+    }
+
+    //TODO: Need to discuss; max= number of teams there are (but there's not point of this challenge, but we dont't know how many teams yet
+    topXTeamsLimit(): number{
+        return 5
+    }
+
+    isTopTeams(): boolean {
+        return this.challengeForm.get('challengeType').value === 'TOP_TEAMS'
+    }
+
+    onSubmit(): void {
+        console.log('submitted')
+
+        const challengeData = CourseEventForm.formatChallengeFormData(
+            this.challengeForm,
+            this.courseId
+        )
+        //TODO: what about the questions in each challenge?
+        if (this.eventId) {
+            this.courseEventService.updateChallenge(challengeData, this.eventId).subscribe(() => {
```

Comment:
The onSubmit is only able to update or create challenges, but the questions within the challenges are not being dealt with. Where should I start for this in the backend?